Data/ByteString/Short.hs

```diff
@@ -67,26 +67,113 @@ module Data.ByteString.Short (
     -- small unpinned strings are allocated in the same way as normal heap
     -- allocations, rather than in a separate pinned area.
 
-    -- * Conversions
-    toShort,
-    fromShort,
+    -- * Introducing and eliminating 'ShortByteString's
+    empty,
+    singleton,
     pack,
     unpack,
+    fromShort,
+    toShort,
 
-    -- * Other operations
-    empty, null, length, index, indexMaybe, (!?),
+    -- * Basic interface
+    snoc,
+    cons,
+    append,
+    last,
+    tail,
+    uncons,
+    head,
+    init,
+    unsnoc,
+    null,
+    length,
 
-    -- ** Encoding validation
+    -- * Encoding validation
     isValidUtf8,
 
+    -- * Transforming ShortByteStrings
+    map,
+    reverse,
+    intercalate,
+
+    -- * Reducing 'ShortByteString's (folds)
+    foldl,
+    foldl',
+    foldl1,
+    foldl1',
+
+    foldr,
+    foldr',
+    foldr1,
+    foldr1',
+
+    -- ** Special folds
+    all,
+    any,
+    concat,
+
+    -- ** Generating and unfolding ByteStrings
+    replicate,
+    unfoldr,
+    unfoldrN,
+
+    -- * Substrings
+
+    -- ** Breaking strings
+    take,
+    takeEnd,
+    takeWhileEnd,
+    takeWhile,
+    drop,
+    dropEnd,
+    dropWhile,
+    dropWhileEnd,
+    breakEnd,
+    break,
+    span,
+    spanEnd,
+    splitAt,
+    split,
+    splitWith,
+    stripSuffix,
+    stripPrefix,
+
+    -- * Predicates
+    isInfixOf,
+    isPrefixOf,
+    isSuffixOf,
+
+    -- ** Search for arbitrary substrings
+    breakSubstring,
+
+    -- * Searching ShortByteStrings
+
+    -- ** Searching by equality
+    elem,
+
+    -- ** Searching with a predicate
+    find,
+    filter,
+    partition,
+
+    -- * Indexing ShortByteStrings
+    index,
+    indexMaybe,
+    (!?),
+    elemIndex,
+    elemIndices,
+    count,
+    findIndex,
+    findIndices,
+
     -- * Low level conversions
-    -- ** Packing 'Foreign.C.String.CString's and pointers
+    -- ** Packing 'CString's and pointers
     packCString,
     packCStringLen,
 
-    -- ** Using ByteStrings as 'Foreign.C.String.CString's
+    -- ** Using ShortByteStrings as 'CString's
```

Comment:
Doesn't this break haddock's hyperlinking on `CString`?!

Reply:
`Using ShortByteStrings as 'CString's` itself is already a link in the generated haddock, so no

Reply:
That doesn't seem right. `CString` is a hyperlinked identifier at https://hackage.haskell.org/package/bytestring-0.11.2.0/docs/Data-ByteString-Short.html#g:8.

Reply:
I think I misunderstood what you were referring to... wasn't even aware we can have links inside links. But from the page you linked they look pretty ugly and break the main link:
![Screenshot_2022-01-24_20-50-09](https://user-images.githubusercontent.com/1241845/150854136-0605ef44-1e64-488a-8bfa-eafe600d20de.png)

So in order to go to the menu, the only word I can click there is **Packing**.

Reply:
Hmm, yeah, that should be improved. Feel free to report this to on the haddock issue tracker. Please fix the link in the `Using ShortByteStrings as CStrings` header though.